synfig-studio/src/gui/app.cpp

```diff
@@ -1348,13 +1346,39 @@ App::get_default_accel_map()
 }
 
 /* === M E T H O D S ======================================================= */
+Glib::RefPtr<App> App::instance()
+{
+	static Glib::RefPtr<studio::App> app_reference = Glib::RefPtr<App>(new App());
+	return app_reference;
+}
+
+void App::run(int& argc, char**& argv)
+{
+	if (argc > 0) {
+		String binary_path = synfig::get_binary_path(String(argv[0]));
```

Comment:
If I understand correctly, there may still be a problem here when handling Unicode paths in Windows. 

From [docs](https://developer-old.gnome.org/glibmm/unstable/classGio_1_1Application.html#a8964856be3086656cc1f933381754d3a):

> Note that on Windows, argc and argv are ignored, and Glib::win32_get_command_line() is called internally (for proper support of Unicode commandline arguments).

The problem is they don't describe **when** this one is called.  Look to the `gapplication.c` code:
```
  if (!G_APPLICATION_GET_CLASS (application)
        ->local_command_line (application, &arguments, &status))
```

So it looks like better process command line arguments by overriding `local_command_line(...)` method.

> If you are interested in doing more complicated local handling of the commandline then you should implement your own Application subclass and override local_command_line(). In this case, you most likely want to return true from your local_command_line() implementation to suppress the default handling. See [gapplication-example-cmdline2.c][gapplication-example-cmdline2] for an example.

Reply:
They already provide the files "opened" in on_open() method.
Could you please test it with files with non-latin names?

Reply:
Oh, maybe you were talking specifically for the line:
` 		String binary_path = synfig::get_binary_path(String(argv[0])); ` right?

OK

Reply:
Yes

Reply:
Fixed, I guess